node/src/isolate/external.cpp

```diff
@@ -229,11 +233,25 @@ struct external_t::inner_t :
     }
 
     void reconnect(const std::error_code& e) {
-        if(session) {
-            COCAINE_LOG_INFO(log, "reconnecting to external isolation daemon");
-            session->detach(e);
-            session.reset();
-            connect();
+        if(!connecting) {
+            connecting = true;
+            if(session) {
+                session->detach(e);
+                session.reset();
+            }
+
+            auto reconnect_timeout_ms = args.as_object().at("reconnect_timeout_ms", 3000u).as_uint();
+            COCAINE_LOG_INFO(log, "queueing reconnect to external isolation daemon after {} ms", reconnect_timeout_ms);
+
+            connect_timer.expires_from_now(boost::posix_time::milliseconds(reconnect_timeout_ms));
+            std::weak_ptr<inner_t> weak_self(shared_from_this());
+            connect_timer.async_wait([=](const std::error_code&) {
```

Comment:
The doc lies, at least for linux.

So, you guarantee that the callback is called while `this` lives? Why do you need weak pointer in this case?